modules/tensor_mechanics/include/materials/ADComputeSmearedCrackingStress.h

```diff
@@ -0,0 +1,158 @@
+                                      const ADReal cracking_stress,
+                                      const ADReal cracking_alpha,
+                                      const ADReal youngs_modulus);
+
+  /**
+   * Get the number of known crack directions. This includes cracks
+   * in prescribed directions (even if not yet active) and active
+   * cracks in other directions.
+   * @return number of known crack directions
+   */
+  virtual unsigned int getNumKnownCrackDirs() const;
+
+  /**
+   * Compute the crack strain in the crack coordinate system. Also
+   * computes the crack orientations, and stores in _crack_rotation.
+   * @param strain_in_crack_dir Computed strains in crack directions
+   */
+  void computeCrackStrainAndOrientation(ADRealVectorValue & strain_in_crack_dir);
+
+  /**
+   * Updates the full stress tensor to account for the effect of cracking
+   * using the provided stresses in the crack directions. The tensor is
+   * rotated into the crack coordinates, modified, and rotated back.
+   * @param tensor Stress tensor to be updated
+   * @param sigma Vector of stresses in crack directions
+   */
+  void updateStressTensorForCracking(ADRankTwoTensor & tensor, const ADRealVectorValue & sigma);
+
+  /**
+   * Check to see whether there was cracking in any diretion in the previous
+   * time step.
+   * @return true if cracked, false if not cracked
+   */
+  bool previouslyCracked();
+
+  /// Enum defining the crack release model
+  const enum class CrackingRelease { abrupt, exponential, power } _cracking_release;
+
+  ///@{ Input parameters for smeared crack models
+
+  /// Ratio of the residual stress after being fully cracked to the tensile
+  /// cracking threshold stress
+  const ADReal _cracking_residual_stress;
+
+  /// Threshold at which cracking initiates if tensile stress exceeds it
+  const ADVariableValue & _cracking_stress;
+
+  /// User-prescribed cracking directions
+  std::vector<unsigned int> _prescribed_crack_directions;
+
+  /// Maximum number of cracks permitted at a material point
+  const unsigned int _max_cracks;
+
+  /// Defines transition to changed stiffness during unloading
+  const ADReal _cracking_neg_fraction;
+
+  /// Controls slope of exponential softening curve
+  const ADReal _cracking_beta;
+
+  /// Controls the amount of shear retained
+  const ADReal _shear_retention_factor;
+
+  /// Controls the maximum amount that the damaged elastic stress is corrected
+  /// to folow the release model during a time step
+  const ADReal _max_stress_correction;
+  ///@}
+
+  //@{ Damage (goes from 0 to 1) in crack directions
+  ADMaterialProperty<RealVectorValue> & _crack_damage;
+  const MaterialProperty<RealVectorValue> & _crack_damage_old;
+  ///@}
+
+  /// Vector of values going from 1 to 0 as crack damage accumulates. Legacy
+  /// property for backward compatibility -- remove in the future.
+  ADMaterialProperty<RealVectorValue> & _crack_flags;
```

Comment:
Doesn't look like this is used at all (the AD property is set up, but I do not see anywhere the AD property is actually gotten)

Reply:
I see, it is output to the exodus file. We might want to consider just removing them and skipping them in the exodiff compare file

Reply:
There are tests checking for those flags specifically...